docs/contributing.md

```diff
@@ -0,0 +1,61 @@
+# Contributing
+This section will demonstrate how to set up a remote development environment. This application utilizes the `devcontainer` directory to build a remote container that will have all necessary dependencies installed to contribute to the **Azure Key Vault Provider for Secret Store CSI Driver**.
+
+### Table of Contents
+- [Prerequisites](#prerequisites)
+- [Clone Repository](#clone-repository)
+- [Open Remote Container](#open-remote-container-in-vs-code)
+- [Debugging Introduction](#debugging-introduction)
+
+## Prerequisites
+1. Visual Studio Code with `Remote - Container` extension enabled.
+2. Azure account and subscription
+3. **optional** Azure Key Vault configured
+
+## Clone Repository
+
+1. Fork and Clone the repository locally
+    - Make sure to change `YOUR_GITHUB_HANDLE` in the git clone command.
+    ```bash
+    # first fork the Repository.
+    # Now clone.
+    $ git clone git@github.com:<YOUR_GITHUB_HANDLE>/secrets-store-csi-driver-provider-azure.git
+    ```
+
+## Open Remote Container in VS Code
+
+1. open VS Code from current path
+    ```bash
+    $ cd secrets-store-csi-driver-provider-azure
+    $ code .
+    ```
+2. In the bottom-left corner of VS Code click on the tag as shown below
+    - If you hover over the tag, it should say "open a remote window"
+
+    ![open a remote window tag](/docs/images/bottom-left.png)
+
+3. Select `Remote-Containers: Reopen in Container` from the drop-down list
+
+    ![Reopen in Container](/docs/images/reopen-container.png)
+
+4. The Azure Key Vault Provider should now be opened inside a Remote Container!
+    - In the bottom-left you should see the tag updated to show: `Dev Container: Secrets Store CSI Driver Provider Azure`
+    - With the terminal open (`ctrl + ``) , on the top-right of the terminal window, you can open up a bash shell by finding the `bash` tab inside the drop-down as pictured below.
+
+    ![remote dev cointainer](/docs/images/container_open.png)
+
+## Debugging Introduction
+
+[Click Here to go to the Debugging Guide](/docs/debug.md#debug-guide)
+
+This guide will walk you through the following:
+
+1. Azure Service Principal Creation
+2. Azure Key Vault Creation, Policy configuration, and adding secrets to the key vault
+3. Set up `debug` Directory
+    - `secrets.env`, `parameters.yaml`, `build-args.sh`, `launch.json`
+4. Running the debugger and breakpoints
+5. Viewing secrets written to `/tmp/secrets`
```

Comment:
Leave this as a simple sentence and link to something like "How to debug the Azure Provider with VS Code and the Remote Container Exntension".